fix(deletions): Try next available group and abort if none found

armenzg opened the pull request:
Fixes [SENTRY-45X2](https://sentry.sentry.io/issues/6731915490/)
Fixes [SENTRY-43KR](https://sentry.sentry.io/issues/6710200116/)

armenzg committed:
fix(deletions): Try next available group and abort if none found

Fixes [SENTRY-45X2](https://sentry.sentry.io/issues/6731915490/)

asottile-sentry reviewed (approved):
![](https://i.fluffy.cc/Xr1jnB3r6dWrfwqPwNdrpnF9zW1lVp0F.gif)

armenzg committed:
Address feedback

armenzg commented:
bugbot run

armenzg commented on tests/sentry/deletions/tasks/test_groups.py:
```diff
@@ -65,3 +68,22 @@ def test_simple(self):
+        group_ids = [group.id, group2.id]
+
+        with self.tasks():
+            group.delete()
```
Makes sense 👍🏻

armenzg committed:
Take chunking into account

The `first_group` used for Sentry `project_id` tagging is selected from the entire `object_ids` list via `Group.objects.filter(id__in=object_ids).first()`. This query lacks an `ORDER BY` clause, so `first_group` can be arbitrary. Additionally, if `object_ids` exceeds `max_batch_size`, `first_group` may belong to a later, un-processed batch. This results in `project_id` tags that do not accurately reflect the groups in the current deletion batch, making debugging difficult.
